### ISSUE TITLE
named parameters using "as" messes with the context

### ISSUE THREAD

**Issue by dontgitit** [jknack/handlebars.java#506]
Hi,

I seem to be having an issue where using `as` messes with non-block helpers. 

When not using `as`, the `lower` helper works properly. 
As soon as `as` is used, `lower` doesn't work. HOWEVER, it can still be resolved via `#lower`

Here's an example...

Template:
```
{{#each things}}
  not using "as": value is {{this}}, key is {{@index}}, hello is {{lower "HELLO"}}
{{/each}}
<br>      
{{#each things as |value key|}}
  using "as": value is {{value}}, key is {{key}}, hello is {{lower "HELLO"}}
{{/each}}
<br>      
{{#each things as |value key|}}
  using "as" and "#": value is {{value}}, key is {{key}}, hello is {{#lower "HELLO"}}{{this}}{{/lower}}
{{/each}}
```

Data:
```
{
  "things":[
    "foo"
  ]
}
```

Result:
```
not using "as": value is foo, key is 0, hello is hello 
using "as": value is foo, key is 0, hello is 
using "as" and "#": value is foo, key is 0, hello is hello
```

P.S. It seems as though maybe it's looking for a value for a variable called `lower`, finding no result and returning it... instead of how it usually works where it will look for helpers with that name.

**Comment by jknack** [comment]
think I fixed this somewhere here: https://github.com/jknack/handlebars.java/commit/30d91be9bc21f79de4ef8b7628b3fdf55e116cb6

bc I can't reproduce it now

**Comment by dontgitit** [comment]
Hmm, must be because we're on 4.0.0... I'll upgrade to the newest version. Thanks!

**Comment by dontgitit** [comment]
Yep, tried out `4.0.5` and it's fixed there, so it was actually fixed somewhere between `4.0.0` and `4.0.5`. Thanks/sorry to bother you with this!

**Comment by jknack** [comment]
no problem, glad is fixed already.
